src/online/unigrid.jl

```diff
@@ -23,48 +23,58 @@ function inducingpoints(
     Z = map(bounds) do lims
         LinRange(lims..., alg.m)
     end
-    return Z
+    proditer = Iterators.product(Z...)
+    return UniformGrid(proditer)
 end
 
 function updateZ!(
     ::AbstractRNG, Z::AbstractVector, alg::UniGrid, X::AbstractVector; kwargs...
 )
-    ndim = length(Z)
+    ndim = length(X[1])
     new_bounds = [extrema(x -> getindex(x, i), X) for i in 1:ndim]
-    map!(Z, Z, new_bounds) do Z_d, new_b
-        x_start = min(Z_d.start, new_b[1]) # Find the new limits
-        x_stop = max(Z_d.stop, new_b[2])
-        return LinRange(x_start, x_stop, alg.m) # readapt bounds
+    Zn = map(Z.proditer.iterators, new_bounds) do old_Z, new_b
+        x_start = min(old_Z.start, new_b[1])
+        x_stop = max(old_Z.stop, new_b[2])
+        return LinRange(x_start, x_stop, alg.m)
     end
+    Z.proditer = Iterators.product(Zn...)
     return Z
 end
 
-function updateZ(rng::AbstractRNG, Z::AbstractVector, alg::UniGrid, X::AbstractVector)
+function updateZ(
+    rng::AbstractRNG, Z::AbstractVector, alg::UniGrid, X::AbstractVector; kwargs...
+)
     Zn = deepcopy(Z)
-    return updateZ!(rng, Zn, alg, X)
+    return updateZ!(rng, Zn, alg, X; kwargs...)
 end
 
 export UniformGrid
-struct UniformGrid{N, T} <: AbstractVector{T} 
-    proditer::Iterators.ProductIterator{NTuple{N, LinRange{T, Int64}}}
+mutable struct UniformGrid{N,T} <: AbstractVector{T}
```

Comment:
It is unfortunate that it has to be mutable, but I guess we can look for a better/different approach later!

Reply:
Yes, sadly the product iterator and all its contents are immutable, so in order to faciliate `updateZ!`, the top struct had to be made mutable.